plugin/adsk/scripts/AETemplateHelpers.py

```diff
@@ -35,4 +45,162 @@ def ProxyShapePayloads(proxyShape, loadAll):
+    #
+    # Return Value:
+    # - True:  The input file path attribute was updated with a new value.
+    # - False: No update to file path attribute was done.
+    #
+    debugMessage('ProxyShapeFilePathChanged(%s, %s)' % (filePathAttr, newFilePath))
+    try:
+        stageName, proxyStage = GetStageFromProxyShapeAttr(filePathAttr)
+
+        # Does the filepath attribute contain a valid file?
+        currFilePath = cmds.getAttr(filePathAttr)
+        dirtyStack = False
+        openFileDialog = False
+        if currFilePath:
+            # We have a current file path, so check if any layer in the layerstack is dirty?
+            dirtyStack = IsProxyShapeLayerStackDirty(proxyStage)
+            if not dirtyStack:
+                openFileDialog = True
+        else:
+            # No current file path, so do we just have an empty stage or was something added to
+            # this anon layer?
+            rootLayer = proxyStage.GetRootLayer()
+            if rootLayer and not rootLayer.empty:
+                dirtyStack = True
+            elif newFilePath is None:
+                debugMessage('  Empty stage, opening file dialog...')
+                openFileDialog = True
+
+        # If we have a dirty layer stack, display dialog to confirm new loading.
+        if dirtyStack:
+            kTitleFormat = mel.eval('getMayaUsdString("kDiscardStageEditsTitle")')
+            kMsgFormat = mel.eval('getMayaUsdString("kDiscardStageEditsLoadMsg")')
+            kYes = mel.eval('getMayaUsdString("kButtonYes")')
+            kNo = mel.eval('getMayaUsdString("kButtonNo")')
+            kTitle = cmds.format(kTitleFormat, stringArg=stageName)
+            # Our title is a little long, and the confirmDialog command is not making the
+            # dialog wide enough to show it all. So by telling the message string to not
+            # word-wrap, the dialog is forced wider.
+            kMsg = "<p style='white-space:pre'>" + cmds.format(kMsgFormat, stringArg=stageName)
+            res = cmds.confirmDialog(title=kTitle,
+                                     message=kMsg, messageAlign='left', 
+                                     button=[kYes, kNo], defaultButton=kYes, cancelButton=kNo, dismissString=kNo,
+                                     icon='warning')
+            if res == kYes:
+                debugMessage('  User confirmed load action, opening file dialog...')
+                openFileDialog = True
+            else:
+                # User said NO to update, so don't proceed any farther.
+                return False
+
+        if openFileDialog and newFilePath is None:
+            # Pop the file open dialog for user to load new usd file.
+            title = mel.eval('getMayaUsdString("kLoadUSDFile")')
+            okCaption = mel.eval('getMayaUsdString("kLoad")')
+            fileFilter = mel.eval('getMayaUsdString("kAllUsdFiles")') + ' (*.usd *.usda *.usdc *.usdz );;*.usd;;*.usda;;*.usdc;;*.usdz';
```

Comment:
Do we have the filter tokens available in Python?  e.g. UsdMayaTranslatorTokens is what we use in c++ when we need filters like this.

Reply:
I see those in the C++ code, but I don't see anywhere they are wrapped into python. I also checked the USD repo and in their python code they just use the actual strings.